testapp/views.py

```diff
@@ -0,0 +1,90 @@
+from pyramid.response import Response
+from pyramid.view import view_config
+import transaction
+from sqlalchemy.exc import DBAPIError
+from .models import (
+    DBSession,
+    Entry,
+)
+from pyramid.httpexceptions import HTTPFound
+from jinja2 import Markup
+import markdown
+from testapp.formclass import EntryForm
+
+
+@view_config(route_name='new', renderer='templates/add.jinja2')
+def new_entry(request):
+    """Create a form page for a new entry."""
+    form = EntryForm(request.POST)
+    if request.method == 'POST' and form.validate():
+        new_entry = Entry(title=form.title.data, text=form.text.data)
+        DBSession.add(new_entry)
+        DBSession.flush()
+        url = request.route_url('entry', id=new_entry.id)
+        return HTTPFound(location=url)
+    return {'form': form}
+
+
+@view_config(route_name='edit', renderer='templates/add.jinja2')
+def edit_entry(request):
+    """Create a form page for an edited entry."""
+    edit_id = request.matchdict['id']
+    edit_entry = DBSession.query(Entry).get(edit_id)
+    # edit_entry.text = render_markdown(edit_entry.text) #trying to get edited entyr to not show html
+    form = EntryForm(request.POST, edit_entry)
+    if request.method == "POST" and form.validate():
+        # edit_entry.text = Markup(edit_entry.text)
+        # edit_entry.text = render_markdown(edit_entry.text)
+        # # edit_entry.text = Markup.striptags(edit_entry.text) #trying to get edited entyr to not show html
+        form.populate_obj(edit_entry)
+        DBSession.add(edit_entry)
+        DBSession.flush()
+        url = request.route_url('entry', id=edit_entry.id)
+        return HTTPFound(location=url)
+    return {'form': form}
+
+
+@view_config(route_name='home', renderer='templates/list.jinja2')
+def home_view(request):
+    """Render home page with database list."""
+    print(request)
+    try:
+        entry_list = DBSession.query(Entry).order_by(Entry.id.desc())
+    except DBAPIError:
+        return Response(conn_err_msg, content_type='text/plain', status_int=500)
+    return {'entry_list': entry_list}
+
+
+@view_config(route_name='entry', renderer='templates/detail.jinja2')
+def entry_view(request):
+    """Render a single page detailed view of an entry."""
+    try:
+        entry_id = request.matchdict['id']
+        single_entry = DBSession.query(Entry).filter(Entry.id == entry_id).first()
+        single_entry.text = render_markdown(single_entry.text)
```

Comment:
This line right here is where your HTML problems come from.  Let's talk through why.

Remember that `single_entry` is a Model.  And that the `text` attribute of that model is a database column.  What that means is that if the value of that attribute is changed, the database session will know it, and when a response is sent to the client, the changes will be committed and persisted to the database.  

In this one line of code, you are setting the `text` attribute of your entry to the result of rendering the markdown that is in the `text` attribute of your entry.  This converts the raw markdown to HTML and then persists the change to the database.  From that point on, your entry will have HTML text, which is why it is showing up that way in the edit form above.  

What would happen if your Entry model had another property (perhaps named `rendered_text`) which returned the rendered HTML from `self.text`?  Then you could use that attribute when displaying the entry, and could preserve the `text` attribute as raw markdown.